### PR TITLE
Fixed corrupted caching issue and lag on navigation animations

### PR DESCRIPTION
**Pull request by yoeven**
Hi there, the pull request fixes two key issues:

1. Previously when the component gets unmounted, it causes the image download to break and save the file regardless of its completion status. Instead, now I use a downloadResumable with a callback to track the progress of its completion and if the image doesn't completely download at the point of unmounting, it would delete the half downloaded file.

2. The component starts the load process of the image on componentDidMount which causes lag to JS-based animations and react-navigation transition. The solution was to use the [InteractionManager.runAfterInteractions](https://reactnative.dev/docs/interactionmanager) function that would start the load process only after any interaction or JS animations are done.

### CODE REVIEW

**Comment by lane-c-wagner** [comment]
Hey @yoeven! Thanks for the PR! A couple things, looks like this PR does more than just those two things:
* Stlying is being changed from `'` to `"`
* The ability to use an <Image> instead of a <BackgroundImage> is being removed.

Can you scope down your PR to ONLY address the scope of what you want to change?

**Comment by yoeven** [comment]
Hi @lane-c-wagner, Thanks for the feedback!

1. I fixed the styling by adding prettier support and also added [expo's eslint config](https://github.com/expo/expo/tree/master/packages/eslint-config-universe) for future PR's so that styling will always stay consistent

2. I'm a little unsure on what you mean by your second point, not really sure what is being removed. You stated that "The ability to use an  instead of a is being removed.", would be great if you could elaborate a little more or provide an example so I can make the right changes. :)

**Comment by lane-c-wagner** [comment]
Hey sorry, I didn't escape point number two properly! It was this: `The ability to use an <Image> instead of a <BackgroundImage> is being removed.`, but now I'm seeing the order was switched, it wasn't removed (I'm on a phone right now haha).

So the only thing I'll ask at this point is that you remove all the styling changes (including prettier). It's typically unwise to try to make styling AND functionality changes in one PR. If you want to make styling changes because you intend to do further work on the project, just open up a separate pull request please :)

**Comment by yoeven** [comment]
Hey there, reverted the prettier changes accordingly and only applied the functionality changes while keeping the styling consistent to the initial repo. I'll send another PR after this gets merge for styling and maybe adding typescript :) Hope its all good now!

File: index.js [hunk 1]
```diff
@@ -1,85 +1,125 @@
...
+      const metadata = await FileSystem.getInfoAsync(filesystemURI);
+      if (metadata.exists) {
+        await FileSystem.deleteAsync(filesystemURI);
+      }
+    }
+  }
+
+  onDownloadUpdate(downloadProgress) {
+    if (downloadProgress.totalBytesWritten >= downloadProgress.totalBytesExpectedToWrite) {
+      console.log('Download completed');
```
Comment (lane-c-wagner) [inline index.js:102]: We shouldn't log success, don't want to clutter the console

**Review by lane-c-wagner — CHANGES_REQUESTED** [review]
last thing is to remove that console.log

File: index.js (hunk 1 above)
Comment (yoeven) [reply index.js]: Alright, removed the unnecessary comments

**Comment by yoeven** [comment]
alright, removed the unnecessary comments

**Comment by yoeven** [comment]
Also if possible could you label this fork with `hacktoberfest-accepted` so that it counts against the participation of the [hacktoberfest ](https://hacktoberfest.digitalocean.com/) event :)

**Comment by lane-c-wagner** [comment]
Absolutely. I'll get we done and published

**Pull request merged**